src/nameable.rb

```diff
@@ -0,0 +1,5 @@
+class Nameable
```

Comment:
```sh
class Nameable
    def correct_name
        raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
    end
end
```

- [ OPTIONAL ] ℹ️ Enhanced error reporting! Self-reference shows the class causing the error, and '#{__method__}' displays the failing method, improving clarity and debuggability. 🚀👍